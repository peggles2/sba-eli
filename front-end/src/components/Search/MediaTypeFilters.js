import React, {Component} from 'react'
import {Form} from 'semantic-ui-react'

export default class MediaTypeFilters extends Component {

  render() {
    //TODO: should this be dynamic?
    const staticMediaFilters = [
      {key: 'video', text: 'Video'},
      {key: 'podcast', text: 'Podcast'},
      {key: 'infographic', text: 'Infographic'},
      {key: 'assessment', text: 'Assessment'},
      {key: 'discussion', text: 'Discussion'},
      {key: 'tools', text: 'Tools'}
    ]

    return staticMediaFilters.map(filter => {
          let selected = this.props.filters && this.props.filters.includes(filter.key)

          return (
              <Form.Checkbox name='mediaType'
                             value={filter.key}
                             key={filter.key}
                             id={'mediatype_filter_' + filter.key}
                             label={{children: filter.text}}
                             defaultChecked={selected}/>
          )
        }
    )
  }
}