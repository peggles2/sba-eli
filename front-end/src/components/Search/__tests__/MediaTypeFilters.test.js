import React from 'react';
import {shallow} from 'enzyme';
import {Form} from "semantic-ui-react";
import MediaTypeFilters from "../MediaTypeFilters";

describe('MediaTypeFilters', () => {
  it('should render 6 <Checkboxes> for 6 media types', () => {
    const wrapper = shallow(<MediaTypeFilters/>);

    expect(wrapper.find(Form.Checkbox).length).toEqual(6);
  });

  it('should render the checkbox with the expected data from the filter', () => {
    const wrapper = shallow(<MediaTypeFilters/>);

    expect(wrapper.find(Form.Checkbox).get(0).key).toEqual('video');
    expect(wrapper.find(Form.Checkbox).get(0).props.value).toEqual('video');
    expect(wrapper.find(Form.Checkbox).get(0).props.id).toEqual('mediatype_filter_video');
    expect(wrapper.find(Form.Checkbox).get(0).props.label.children).toEqual('Video');
  });

});
