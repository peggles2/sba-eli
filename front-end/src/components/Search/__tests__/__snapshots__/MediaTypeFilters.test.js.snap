// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaTypeFilters should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaTypeFilters />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "video",
    "nodeType": "function",
    "props": Object {
      "as": [Function],
      "control": [Function],
      "id": "mediatype_filter_video",
      "label": Object {
        "children": "Video",
      },
      "name": "mediaType",
      "value": "video",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "video",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_video",
        "label": Object {
          "children": "Video",
        },
        "name": "mediaType",
        "value": "video",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "podcast",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_podcast",
        "label": Object {
          "children": "Podcast",
        },
        "name": "mediaType",
        "value": "podcast",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "infographic",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_infographic",
        "label": Object {
          "children": "Infographic",
        },
        "name": "mediaType",
        "value": "infographic",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "assessment",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_assessment",
        "label": Object {
          "children": "Assessment",
        },
        "name": "mediaType",
        "value": "assessment",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "discussion",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_discussion",
        "label": Object {
          "children": "Discussion",
        },
        "name": "mediaType",
        "value": "discussion",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "tools",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "mediatype_filter_tools",
        "label": Object {
          "children": "Tools",
        },
        "name": "mediaType",
        "value": "tools",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
